chipsec/hal/iobar.py

```diff
@@ -88,6 +88,10 @@ def get_IO_BAR_base_address( self, bar_name ):
             # this method is not preferred
             base = self.cs.pci.read_word( int(bar['bus'],16), int(bar['dev'],16), int(bar['fun'],16), int(bar['reg'],16) )
 
+        if 'fixed_address' in bar:
+            base = int(bar['fixed_address'],16)
+            if logger().VERBOSE: logger().log('[iobar] Using fixed address for {}: 0x{:016X}'.format(bar_name, base))
+
```

Comment:
You should only override the base address if you were not able to read it from a register or PCI configuration space. I would expect a check for 0xFFFF as well as checking for the fixed address attribute.

Reply:
Ok, how about the latest commit?